### PR TITLE
f-feature-management@v0.6.0 - Add Vue service and store with tests + reorganise module

### PR DESCRIPTION
**Pull request by jnewc**
This encapsulates all Vue and VueX related logic and setup for feature management in a new service (FeaturesService) to allow cleaner injection of the service as a plugin.

All Vue-specific code is in `src/vue`. The Vue-independent parts of the library are in `src/lib`. 

---

- [x] Unit tests have been created and updated

### CODE REVIEW

File: packages/services/f-feature-management/src/vue/store/features.module.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import SET_FM_CONFIG_JSON from './mutation.types';
+
+export default {
+    state: () => ({
+        configJson: {}
+    }),
+
+    actions: {
+        getConfigJson: ({ state }) => state.configJson,
```
Comment (xander-marjoram) [inline packages/services/f-feature-management/src/vue/store/features.module.js]: Not sure this is necessary, you should just be able to read `configJson` directly via `mapState`.

File: packages/services/f-feature-management/src/vue/services/features.service.js [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+    return settings;
+}
+
+function createLoggerWrapper (store, logger) {
+    const tags = ['feature-management'];
+    return {
+        logInfo: message => logger && logger.logInfo(message, store, { tags }),
+        logWarn: message => logger && logger.logWarn(message, store, { tags }),
+        logError: message => logger && logger.logError(message, store, { tags })
```
Comment (DamianMullins) [inline packages/services/f-feature-management/src/vue/services/features.service.js]: Should be able to use optional chaining here

```suggestion
        logInfo: message => logger?.logInfo(message, store, { tags }),
        logWarn: message => logger?.logWarn(message, store, { tags }),
        logError: message => logger?.logError(message, store, { tags })
```

File: packages/services/f-feature-management/src/vue/services/features.service.js [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    };
+}
+
+/**
+ * Vue-friendly service for configuring and providing a Feature Management plugin with its own VueX store
+ */
+export default class FeaturesService {
+    constructor (store, {
+        httpClient, cookies, analytics, logger
```
Comment (DamianMullins) [inline packages/services/f-feature-management/src/vue/services/features.service.js]: NIT: Can you wrap these into individual lines for readability, please?

File: packages/services/f-feature-management/test/vue/features.service.test.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+};
+
+function createSUT () {
+    return new FeaturesService(mockStore, {
+        httpClient: mockHttpClient,
+        cookies: mockCookies,
+        analytics: mockAnalytics,
+        logger: mockLogger
+    });
+}
```
Comment (DamianMullins) [inline packages/services/f-feature-management/test/vue/features.service.test.js]: If you're not referencing `this` anywhere you could convert this to an arrow function to keep it consistent with other function definitions?

```suggestion
const createSUT = () => new FeaturesService(mockStore, {
    httpClient: mockHttpClient,
    cookies: mockCookies,
    analytics: mockAnalytics,
    logger: mockLogger
});
```

Comment (DamianMullins) [reply packages/services/f-feature-management/test/vue/features.service.test.js]: Although looking through this project it looks like a lot of the functions are defined using the `function` keyword... I thought we had a lint rule that pops a warning when doing that if you're not using `this` 🤔

**Review by xander-marjoram — APPROVED** [review]

**Review by DamianMullins — APPROVED** [review]

**Pull request merged**